app/code/community/Payright/Payright/controllers/PaymentController.php

```diff
@@ -135,7 +139,7 @@ public function cancelAction()
             $transactionStatus = $result->prtransactionStatus;
             $planid            = $result->planId;
             if ($transactionStatus != "approved" && $transactionStatus != "Declined") {
-                $helper = Mage::helper('payright')->planStatusChange($planid);
```

Comment:
Check if the "approved" is correct, shouldn't it be Approved , is this tied with the Sugar Responses ?

Reply:
This has been used to Cancelled the plan. I just refactor the planStatusChange method to use for activation as well.